### PR TITLE
feat: frames

### PR DESCRIPTION
**Pull request by innocenzi**
This pull request implements Frames, a way to dynamically load a component and its properties inside a page component.

```php
use function Hybridly\frame;

final class ShowProfileCardController
{
    public function __invoke(User $user)
    {
        return frame('users.profile-card', [
            'user' => UserProfileData::from($user),
        ]);
    }
}
```

```vue
<hybrid-frame :href="route('users.show-profile-card', { user: $props.user.id })" />
```

The component name can either be returned by the server (as the first parameter to the `frame` function) or defined by the `component` property of `hybrid-frame`. Alternatively, a slot can be used:

```vue
<hybrid-frame
	v-slot="{ user }: { user: App.Data.UserProfileData }"
	:href="route('users.show-profile-card')"
>
  {{ user }}
</hybrid-frame>
```

Note that frames are loaded when mounted or when the `href` changes, so this might not be compatible with SSR once it lands.

---

https://github.com/hybridly/hybridly/assets/16060559/5f873b66-60f2-4d60-86a9-a57be9810b90

<details>
<summary>Front-end code for the example above</summary>

```vue
<script setup lang="ts">
import { RouterLink } from 'hybridly/vue'
import { HoverCardArrow, HoverCardContent, HoverCardPortal, HoverCardRoot, HoverCardTrigger } from 'radix-vue'

const $props = withDefaults(defineProps<{
	user: App.Data.UserData | App.Data.UserProfileData
	size?: 'xs' | 'sm' | 'lg' | 'xl' | '2xl'
	darkerBackground?: boolean
}>(), {
	size: 'sm',
	darkerBackground: false,
})

const hoverState = ref(false)
</script>

<template>
	<hover-card-root v-model:open="hoverState" :open-delay="200">
		<!-- Trigger -->
		<hover-card-trigger
			:as="RouterLink"
			class="inline-block cursor-pointer rounded-full shadow-[hsl(206_22%_7%_/_35%)_0px_10px_38px_-10px,hsl(206_22%_7%_/_20%)_0px_10px_20px_-15px] outline-none focus:shadow-[0_0_0_2px_white]"
			:href="route('users.show', { user: user.id })"
			target="_blank"
			rel="noreferrer noopener"
		>
			<avatar :user="user" :size="size" :darker-background="darkerBackground" />
		</hover-card-trigger>

		<!-- Content -->
		<hover-card-portal>
			<hover-card-content
				class="z-50 data-[side=bottom]:animate-slideUpAndFade data-[side=right]:animate-slideLeftAndFade data-[side=left]:animate-slideRightAndFade data-[side=top]:animate-slideDownAndFade w-[300px] rounded-md bg-white p-5 shadow-[hsl(206_22%_7%_/_35%)_0px_10px_38px_-10px,hsl(206_22%_7%_/_20%)_0px_10px_20px_-15px] data-[state=open]:transition-all"
				:side-offset="5"
				:avoid-collisions="true"
			>
				<hybrid-frame component="users.profile-card" :href="route('users.show-profile-card', { user: $props.user.id })" />
				<hover-card-arrow class="fill-white" :width="8" />
			</hover-card-content>
		</hover-card-portal>
	</hover-card-root>
</template>
```

</details>

### CODE REVIEW

**Comment by ninjaparade** [comment]
@innocenzi I approved this PR, feel free to merge it.

**Comment by Tofandel** [comment]
I would just add a section in the doc to explain the difference with the `Dialogs` feature which also creates a backlink to that page and do the same on the `Dialogs` documentation

File: docs/guide/frames.md [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+# Frames
+
+## Overview
+
+Frames are a way to dynamically load a component and its properties from the server, similarly to page components.
+
+They are defined using the `hybrid-frame` component, and their properties are loaded when mounted through an AJAX request using the `href` property.
+
+:::info Experimental
+This function has not been dogfed yet and is considered experimental. Its API may change at any time. Feel free to give feedback on our Discord server.
```
Comment (Tofandel) [inline docs/guide/frames.md:14]:
```suggestion
This feature has been recently introduced and is considered experimental. Its API may change at any time. Feel free to give feedback on our Discord server.
```

File: docs/guide/frames.md [hunk 2]
```diff
@@ -0,0 +1,102 @@
+---
+outline: 'deep'
+---
+
+# Frames
+
+## Overview
+
+Frames are a way to dynamically load a component and its properties from the server, similarly to page components.
```
Comment (Tofandel) [inline docs/guide/frames.md:9]:
```suggestion
Frames are a way to dynamically load a component and its properties from the server, similarly to [dialogs](dialogs.html), but unlike them, they are not standalone pages and do not have a dedicated url for the end user and do not require a base, so they can be loaded in any page.

```

Not sure how links work in the doc but something like this

**Comment by pedro-santiago** [comment]
@Tofandel that's a great improvement on how extensible Hibridly is. Great idea!

**Pull request closed without merging**